lib/jobs/validate-ssh.js

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+module.exports = validationJobFactory;
+
+di.annotate(validationJobFactory, new di.Provide('Job.Ssh.Validation'));
+di.annotate(validationJobFactory, new di.Inject(
+    'Job.Base',
+    'Util',
+    'Logger',
+    'Assert',
+    'Promise',
+    'Services.Waterline',
+    'ssh',
+    '_'
+));
+
+function validationJobFactory(
+    BaseJob,
+    util,
+    Logger,
+    assert,
+    Promise,
+    waterline,
+    ssh,
+    _
+) {
+    var logger = Logger.initialize(validationJobFactory);
+    function ValidationJob(options, context, taskId) {
+        ValidationJob.super_.call(this, logger, options, context, taskId);
+        assert.string(this.context.target);
+        this.nodeId = this.context.target;
+        assert.ok(options.users || this.context.users);
+        this.timeout = options.timeout || 10000;
+        this.retries = options.retries || 2;
+        this.backoffDelay = options.backoffDelay || 1000;
+        this.users = _.compact([].concat(options.users).concat(this.context.users));
+    }
+    util.inherits(ValidationJob, BaseJob);
+
+    ValidationJob.prototype._run = function run() {
+        var self = this;
+        return waterline.lookups.findByTerm(self.nodeId)
+        .then(function(lookups) {
+            return self.testCredentials(lookups, self.users, self.retries, self.backoffDelay);
+        })
+        .then(function(sshSettings) {
+            return waterline.nodes.updateByIdentifier(
+                self.nodeId,
+                {sshSettings: sshSettings}
```

Comment:
yes, once we find valid settings we also want to make them available on the node document.